### PR TITLE
Add parsing support for revoke and implementation for hive connector

### PR DESCRIPTION
**Pull request by amrutagokhale**
@dain Please review.

### CODE REVIEW

**Comment by facebook-github-bot-1** [comment]
Thank you for your pull request and welcome to our community. We require contributors to sign our Contributor License Agreement, and we don't seem to have you on file. In order for us to review and merge your code, please sign up at https://code.facebook.com/cla - and if you have received this in error or have any questions, please drop us a line at cla@fb.com. Thanks!

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java [hunk 1]
```diff
@@ -1305,6 +1305,20 @@ public void grantTablePrivileges(ConnectorSession session, SchemaTableName schem
     }
 
     @Override
+    public void revokeTablePrivileges(ConnectorSession session, SchemaTableName schemaTableName, Set<Privilege> privileges, String grantee, boolean grantOption)
+    {
+        String schemaName = schemaTableName.getSchemaName();
+
```
Comment (martint) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: unnecessary empty line

File: presto-main/src/main/java/com/facebook/presto/execution/RevokeTask.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+        Set<Privilege> privileges;
+        if (statement.getPrivileges().isPresent()) {
+            privileges = statement.getPrivileges().get().stream()
+                    .map(privilege -> parsePrivilege(statement, privilege))
+                    .collect(toImmutableSet());
+        }
+        else {
+            // All privileges
+            privileges = ImmutableSet.copyOf(Privilege.values());
```
Comment (martint) [inline presto-main/src/main/java/com/facebook/presto/execution/RevokeTask.java]: You can do `EnumSet.allOf(Privilege.class)`;

**Comment by martint** [comment]
@dain, I reviewed the grammar and task bits. Can you take a look at the access checks and connector changes?

File: presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java (hunk 1 above)
Comment (amrutagokhale) [inline presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java]: removed.

File: presto-main/src/main/java/com/facebook/presto/execution/RevokeTask.java (hunk 2 above)
Comment (amrutagokhale) [inline presto-main/src/main/java/com/facebook/presto/execution/RevokeTask.java]: Modified. Also added change in GrantTask.java.

**Comment by amrutagokhale** [comment]
Updated.

File: presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java [hunk 3]
```diff
@@ -937,6 +937,52 @@ public void grantTablePrivileges(String databaseName, String tableName, String g
...
+                                principalType = ROLE;
+                            }
+                            else {
+                                principalType = USER;
+                            }
+
+                            ImmutableList.Builder<HiveObjectPrivilege> privilegeBagBuilder = ImmutableList.builder();
+                            for (PrivilegeGrantInfo privilegeGrantInfo : privilegeGrantInfoSet) {
+                                privilegeBagBuilder.add(
+                                        new HiveObjectPrivilege(new HiveObjectRef(HiveObjectType.TABLE, databaseName, tableName, null, null),
```
Comment (dain) [inline presto-hive/src/main/java/com/facebook/presto/hive/metastore/CachingHiveMetastore.java]: place the first argument on a new line

**Comment by dain** [comment]
Merged, thanks!

**Pull request closed without merging**